### PR TITLE
fix: re-render on booker

### PR DESCRIPTION
**Pull request by sean-brydon**
Moves duration to the useEvent call which pulls from trpc cache
CleanShot 2023-10-24 at 15 27 50
CleanShot 2023-10-24 at 15 27 54
Network call down to 1 from 2 as reported in thread: https://threads.com/messages/648066a6-127a-4519-b3cf-c9a2d7ba9c5a?messageID=0ee75155-786f-485c-9e33-1626b0018f01
CleanShot 2023-10-24 at 15 28 07

### CODE REVIEW

**Comment by sean-brydon** [comment]
Current dependencies on/for this PR:
* main
  * **PR #12058** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/12058" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/calcom/cal.com/12058?utm_source=stack-comment).

File: packages/features/bookings/Booker/components/EventMeta.tsx [hunk 1]
```diff
@@ -38,13 +37,6 @@ export const EventMeta = () => {
   const isEmbed = useIsEmbed();
   const hideEventTypeDetails = isEmbed ? embedUiConfig.hideEventTypeDetails : false;
 
-  useEffect(() => {
-    if (!selectedDuration && event?.length) {
-      setSelectedDuration(event.length);
-    }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [event?.length, selectedDuration]);
```
Comment (sean-brydon) [inline packages/features/bookings/Booker/components/EventMeta.tsx:46]: Was causing the whole booker to re-render. not ideal. Shallow doesnt work as expected here

File: packages/features/bookings/components/AvailableTimes.tsx [hunk 2]
```diff
@@ -51,9 +52,9 @@ const SlotItem = ({
   const overlayCalendarToggled =
     getQueryParam("overlayCalendar") === "true" || localStorage.getItem("overlayCalendarSwitchDefault");
   const [timeFormat, timezone] = useTimePreferences((state) => [state.timeFormat, state.timezone]);
-  const selectedDuration = useBookerStore((state) => state.selectedDuration);
   const bookingData = useBookerStore((state) => state.bookingData);
   const layout = useBookerStore((state) => state.layout);
+  const { data: event } = useEvent();
```
Comment (sean-brydon) [inline packages/features/bookings/components/AvailableTimes.tsx:57]: Pulll from cached value

File: packages/features/bookings/components/AvailableTimes.tsx [hunk 3]
```diff
@@ -67,11 +68,12 @@ const SlotItem = ({
...
   const offset = (usersTimezoneDate.utcOffset() - nowDate.utcOffset()) / 60;
 
-  const { isOverlapping, overlappingTimeEnd, overlappingTimeStart } = useCheckOverlapWithOverlay(
-    computedDateWithUsersTimezone,
-    selectedDuration,
-    offset
-  );
+  const { isOverlapping, overlappingTimeEnd, overlappingTimeStart } = useCheckOverlapWithOverlay({
+    start: computedDateWithUsersTimezone,
+    selectedDuration: event?.length ?? 0,
```
Comment (sean-brydon) [inline packages/features/bookings/components/AvailableTimes.tsx:73]: We should never hit event.length being null here -> we pull directly from cache from the inital call from loading booker

File: packages/features/bookings/lib/useCheckOverlapWithOverlay.tsx [hunk 4]
```diff
@@ -9,7 +9,15 @@ function getCurrentTime(date: Date) {
...
-export function useCheckOverlapWithOverlay(start: Dayjs, selectedDuration: number | null, offset: number) {
+export function useCheckOverlapWithOverlay({
+  start,
+  selectedDuration,
+  offset,
+}: {
+  start: Dayjs;
+  selectedDuration: number | null;
+  offset: number;
+}) {
```
Comment (sean-brydon) [inline packages/features/bookings/lib/useCheckOverlapWithOverlay.tsx:20]: NIt: Cleaner than previous approach

**Review by keithwillcode — APPROVED** [review]
Code looks good. Testing on preview now.

File: packages/features/bookings/components/AvailableTimes.tsx (hunk 3 above)
Comment (emrysal) [reply packages/features/bookings/components/AvailableTimes.tsx:73]: is 0 a safe response? I'm not sure what our slot logic does if it has a duration of 0

Comment (sean-brydon) [reply packages/features/bookings/components/AvailableTimes.tsx:73]: 0 is just for calculating overlap with overlay calendar - will just show as everything has available until something loads

**Comment by keithwillcode** [comment]
Can confirm `getSchedule` is not called twice now but cannot fully test though due to tRPC issues calling that endpoint.

**Review by zomars — APPROVED** [review]

**Pull request merged**